code/modules/organs/subtypes/standard.dm

```diff
@@ -43,6 +43,21 @@
 	gendered_icon = 1
 	augment_limit = 3
 
+	var/obj/item/storage/internal/tail/tail_storage
+
+/obj/item/organ/external/groin/Initialize()
+	. = ..()
+	tail_storage = new /obj/item/storage/internal/tail(src)
+
+/obj/item/organ/external/groin/Destroy()
+	QDEL_NULL(tail_storage)
+	return ..()
+
+/obj/item/organ/external/groin/removed()
+	. = ..()
+	for(var/thing in tail_storage)
+		tail_storage.remove_from_storage(thing, get_turf(src))
+
```

Comment:
It seems a bit suspect that this doesn't check if the species actually has a tail.